packages-exp/auth-exp/src/core/auth/auth_impl.ts

```diff
@@ -211,7 +211,22 @@ export class AuthImpl implements Auth, _FirebaseService {
     this._deleted = true;
   }
 
-  async updateCurrentUser(user: externs.User | null): Promise<void> {
+  async updateCurrentUser(userExtern: externs.User | null): Promise<void> {
+    // The public updateCurrentUser method needs to make a copy of the user,
+    // and also needs to verify that the app matches
+    const user = userExtern as User | null;
+    assert(
+      !user || user.auth.name === this.name,
+      AuthErrorCode.ARGUMENT_ERROR,
+      {
+        appName: this.name
+      }
+    );
+
+    return this._updateCurrentUser(user && user._clone());
```

Comment:
user?._clone()

Reply:
That types out to `User|undefined` but the method wants `User|null`. So it's either `user?._clone() || null` or `user && user._clone()`. The latter seems cleaner to me so I'm keeping it the way it's written